test/new-e2e/tests/installer/all_packages_test.go

```diff
@@ -145,79 +145,63 @@ func (s *packageBaseSuite) ProvisionerOptions() []awshost.ProvisionerOption {
-			"apm_dd_url": s.Env().FakeIntake.URL,
-		}
+func (s *packageBaseSuite) RunInstallScriptWithError(params ...string) error {
+	// FIXME: use the official install script
+	_, err := s.Env().RemoteHost.Execute(fmt.Sprintf(`%s bash -c "$(curl -L https://storage.googleapis.com/updater-dev/install_script_agent7.sh)"`, strings.Join(params, " ")), client.WithEnvVariables(installScriptEnv(s.arch)))
+	if err != nil {
+		return err
 	}
-	rawDatadogConfig, err := yaml.Marshal(datadogConfig)
-	require.NoError(s.T(), err)
-	s.Env().RemoteHost.MustExecute(fmt.Sprintf("echo '%s' | sudo tee /etc/datadog-agent/datadog.yaml", string(rawDatadogConfig)))
-	_, err = s.Env().RemoteHost.Execute("sudo datadog-installer version")
 	// Right now the install script can fail installing the installer silently, so we need to do this check or it will fail later in a way that is hard to debug
+	_, err = s.Env().RemoteHost.Execute("sudo datadog-installer version")
+	return err
+}
+
+func (s *packageBaseSuite) RunInstallScript(params ...string) {
+	err := s.RunInstallScriptWithError(params...)
 	require.NoErrorf(s.T(), err, "installer not properly installed. logs: \n%s\n%s", s.Env().RemoteHost.MustExecute("cat /tmp/datadog-installer-stdout.log"), s.Env().RemoteHost.MustExecute("cat /tmp/datadog-installer-stderr.log"))
 }
 
-func (s *packageBaseSuite) InstallAgentPackage() {
-	s.Env().RemoteHost.MustExecute(`sudo -E datadog-installer install oci://669783387624.dkr.ecr.us-east-1.amazonaws.com/agent-package:`+fmt.Sprintf("pipeline-%v", os.Getenv("CI_PIPELINE_ID")), client.WithEnvVariables(s.env))
-	s.Env().RemoteHost.MustExecute(`timeout=30; unit=datadog-agent.service; while ! systemctl is-active --quiet $unit && [ $timeout -gt 0 ]; do sleep 1; ((timeout--)); done; [ $timeout -ne 0 ]`)
+func envForceInstall(pkg string) string {
+	return "DD_INSTALLER_DEFAULT_PKG_INSTALL_" + strings.ToUpper(strings.ReplaceAll(pkg, "-", "_")) + "=true"
 }
 
-// TODO: This is a hack to install a working version of apm-inject
-func (s *packageBaseSuite) InstallInjectorPackageTemp() {
-	s.Env().RemoteHost.MustExecute("sudo -E datadog-installer install oci://669783387624.dkr.ecr.us-east-1.amazonaws.com/apm-inject-package:pipeline-34717565", client.WithEnvVariables(s.env))
+func (s *packageBaseSuite) Purge() {
+	s.Env().RemoteHost.MustExecute("sudo apt-get remove -y --purge datadog-installer || sudo yum remove -y datadog-installer")
 }
 
-// TODO: This is a hack to install a working version of apm-inject
-func (s *packageBaseSuite) InstallInjectorPackageTempWithError() error {
-	s.Env().RemoteHost.MustExecute(`echo "DD_APM_RECEIVER_SOCKET=/var/tmp/apm.socket" | sudo tee -a /etc/environment`)
+// setupGlobalEnv sets up the global environment variables for the agent processes
+// This is done with SystemD environment files overrides to avoid having to touch the agent configuration files
+// and potentially interfere with the tests.
+func (s *packageBaseSuite) setupGlobalEnv() {
```

Comment:
this method shouldn't be required anymore, at least everything touching the injector